src/Microsoft.OData.Core/ODataAsynchronousReader.cs

```diff
@@ -31,6 +31,8 @@ public sealed class ODataAsynchronousReader
         /// </summary>
         private readonly IServiceProvider container;
 
+        private byte[] byteBuffer = new byte[1];

```

Comment:
Why is the buffer size 1?

Reply:
@KenitoInc @ElizabethOkerio When reading the first line plus the headers lines of the raw input payload, we perform the reading one character at a time to ensure their validity
```
HTTP/1.1 200 OK
Content-Type: application/json
OData-Version: 4.0
```
For the synchronous API, we make use of the framework's `ReadByte` method defined in the `Stream` class. For the asynchronous API, we achieve the same using the `ReadAsync` method, reading a single byte at a time. That's what we're using this single-byte array for. We are declaring and initializing it here to avoid unnecessary allocations
```csharp
private async Task<int> ReadByteAsync()
{
    if (await this.rawInputContext.Stream.ReadAsync(byteBuffer, 0, 1).ConfigureAwait(false) != 0)
    {
        return byteBuffer[0];
    }

    return -1;
}
```

Reply:
@KenitoInc this is also so that it can be used in closures, not sure if that is correct @gathogojr

Reply:
I'm using it here https://github.com/OData/odata.net/pull/2364/files#diff-7ed6791834f0962c4865ca7047d3fc29b976468cfcc949afbef91b87f6662613R455

Reply:
@habbes Our offline discussion about the reasons we do it refers